AstSemantics.md

```diff
@@ -35,9 +35,14 @@ a trap occurs.
 ## Local and Memory types
 
 Individual storage locations in WebAssembly are typed, including global
-variables, local variables, and parameters. The heap itself is not typed, but
-all accesses to the heap are annotated with a type. The legal types for
-global variables and heap accesses are called *Memory types*.
+variables, local variables, and parameters. The main storage of a wasm module, 
+called the *Linear memory*, is a contiguous, byte-addressable range of memory
+spanning from offset `0` to `memory_size`. The linear memory can be considered 
+to be a untyped array of bytes which is stored separately from the global variables.
+The linear memory is sandboxed; it does not alias the execution engine's internal
+data structures, the stack, or other process memory.
```

Comment:
I changed it to "execution stack" to be more clear. Emscripten's shadow stack for addressable local variables is an application detail, IMO, so I wanted to avoid discussing it here.